apps/passport-client/src/service-worker.ts

```diff
@@ -10,36 +10,56 @@ const impermanentCache = [
   // "/",
   // "/favicon.ico",
   // "/index.html",
-  // "/global-pcdpass.css",
   // "/global-zupass.css",
-  // "/js/index.js",
+  // "/js/index.js"
 ];
 
-async function addResourcesToCache(resources: string[]): Promise<void> {
+const permanentCache = [
+  "/semaphore-artifacts/16.wasm",
+  "/semaphore-artifacts/16.zkey",
+  "/artifacts/zk-eddsa-event-ticket-pcd/circuit.wasm",
+  "/artifacts/zk-eddsa-event-ticket-pcd/circuit.zkey",
+  "/fonts/IBMPlexSans-Regular.woff",
+  "/fonts/IBMPlexSans-Medium.woff",
+  "/fonts/IBMPlexSans-Light.woff",
+  "/fonts/IBMPlexSans-ExtraLight.woff"
+];
+
+function requestToItemCacheKey(request: Request): string {
+  // This may need updating if the list above ever includes more complex
+  // URLs, with parameters, or across domains.
+  return new URL(request.url).pathname;
+}
+
+async function addResourcesToCache(): Promise<void> {
   const cache = await caches.open("v1");
 
+  const keys = await cache.keys();
+
+  // Delete all existing cache entries not in our "permanent" list.
+  // This eliminates any stale entries from old service workers which
+  // used a different list.
   await Promise.all(
-    ["/", "/index.html", "/js/index.js"].map((item) => cache.delete(item))
+    keys.map((request: Request) => {
+      const urlKey = requestToItemCacheKey(request);
+      if (!permanentCache.includes(urlKey)) {
+        console.log(`[SERVICE_WORKER] discarding ${urlKey}`);
+        cache.delete(request);
```

Comment:
```suggestion
        return cache.delete(request);
```

?

Reply:
Ooh, tricky.  Promises confuse me with their many different usage models, but yes there's a promise here which never gets checked.  It seems to work fine, presumably because it doesn't really matter if the app continues before delete happens, but it's a race.  I think in this case an await will better express the intent.